module-03-react/src/App.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useRef, useState } from "react";
+import "./App.css";
+import Gallery from "./components/Gallery";
+import SideMenu from "./components/SideMenu";
+import TopMenu from "./components/TopMenu";
+import BottomBtn from "./components/BottomBtn";
+import Modal from "./components/Modal";
+
+
+function App() {
+  const galleryRef = useRef();
+  const [url, setUrl] = useState("people/");
+  const [search, setSearch] = useState('');
+  const [openModal, setOpenModal] = useState(false);
+  const [choosedCard, setChoosedCard] = useState();
+  const [img, setImg] = useState([]);
+
+
+  const loadMore = () => {
+    if (galleryRef.current) {
+      galleryRef.current.loadMore();
```

Comment:
This line uses an imperative approach to trigger the loadMore method on the Gallery component through a ref, which goes against React's declarative programming model. Instead of directly calling methods on child components, consider lifting the pagination state up to the App component and passing it down as props. This would improve component decoupling, make the code easier to test, and better align with React's unidirectional data flow pattern